packages/semaphore-signature-pcd/src/SemaphoreSignaturePCD.ts

```diff
@@ -63,12 +64,24 @@ export interface SemaphoreSignaturePCDClaim {
    * Stringified `BigInt`.
    */
   identityCommitment: string;
-}
 
-export interface SemaphoreSignaturePCDProof {
-  proof: FullProof;
+  /**
+   * Stringified `BigInt`.
+   */
+  nullifierHash: string;
+
+  /**
+   * Stringified `BigInt`.
+   */
+  externalNullifier: string;
```

Comment:
`externalNullifier` should not be able to be set for signatures. You can think of the `externalNullifier` as a group identifier that is hashed along your private key to come up with a `nullifier` for a user (https://github.com/semaphore-protocol/semaphore/blob/main/packages/circuits/scheme.png). The nullifier is anonymous and separate from your identityCommitment. However, if you allow this field to be set, then I can request a signature from you with the same externalNullifier as a group we're both in, and then I can figure out your nullifier and break your anonymity. This is because the nullifier is outputted by the Semaphore circuits on default.

We should set it to some fixed but basically random value (previously, it was the identityCommitment of the user). It shouldn't be something like "1" because there will likely be (noob) groups set up with that as an externalNullifier and we shouldn't break the anonymity of those people with this other PCD. If we decide on the identityCommitment being the externalNullifier, we should also have a check in the SemaphoreGroupPCD to make sure those PCDs aren't ever generated with externalNullifier == identityCommitment, which would break those user's anonymity if an adversary also had a signature from them.

Reply:
ok. i'll set it to some hardcoded random big number.